core/metacling/src/TClingCallFunc.cxx

```diff
@@ -229,6 +229,19 @@ void TClingCallFunc::collect_type_info(QualType &QT, ostringstream &typedefbuf,
    GetTypeAsString(QT, type_name, C, Policy);
 }
 
+static bool IsCopyConstructorDeleted(QualType QT)
+{
+   if (CXXRecordDecl *RD = QT->getAsCXXRecordDecl()) {
+      for (auto *Ctor : RD->ctors()) {
+         if (Ctor->isCopyConstructor()) {
+            return Ctor->isDeleted();
+         }
+      }
+   }
```

Comment:
```suggestion
      CXXRecordDecl *RD = QT->getAsCXXRecordDecl();
      if (!RD)
        return false; // Not sure what we should return here actually
        
      if (!(RD = RD->getDefinition()))
        return false; // ?

      if (RD->hasSimpleCopyConstructor())
        return true;
      // Maybe check for trivial copy ctor...
 
      for (auto *Ctor : RD->ctors()) {
         if (Ctor->isCopyConstructor()) {
            return Ctor->isDeleted();
         }
   }
```

Looks better. I think we can guard the iteration with simpler, less expensive checks. The iteration also deserializes in memory...

Reply:
For non-`CXXREcordDecl`, we assume it's a simple type and that it's copyable. The check for `hasSimpleCopyConstructor()` makes sense to me, I will add that.

Reply:
The question is whether we should add a check for hasTrivialCopyCtor

Reply:
Turns out no, because `= delete` is trivial. Same for `hasUserDeclaredCopyCtor()`.